client/src/driver/BitbakeRecipeScanner.ts

```diff
@@ -9,14 +9,37 @@ import { logger } from '../lib/src/utils/OutputLogger'
 import { runBitbakeTask } from '../ui/BitbakeCommands'
 import { type BitbakeCustomExecution, type BitbakeTaskProvider } from '../ui/BitbakeTaskProvider'
 import { RequestMethod, type RequestParams } from '../lib/src/types/requests'
+import { extractRecipeName } from '../lib/src/utils/files'
 
-export class BitbakeRecipeScanner {
+type ServerScanRequestListener = (recipeName: string) => Promise<void>
+export class BitbakeRecipeScanner implements vscode.Disposable {
   static readonly taskName = 'Bitbake: Scan recipe env'
 
   private _languageClient: LanguageClient | undefined
   private _pendingRecipeScanTasks: vscode.Task | null = null
 
-  private readonly serverScanRequest = new vscode.EventEmitter<vscode.Uri>()
+  private readonly serverScanRequest = new vscode.EventEmitter<vscode.Uri | undefined>()
+  private readonly serverScanRequestListeners: ServerScanRequestListener[] = []
+
+  private readonly disposables: vscode.Disposable[] = []
```

Comment:
You should let the class caller do the disposal when they add an event. It's not this class responsibility. Also I don't understand why this event is declared here since it's not fired from this class.

Reply:
Are you suggesting to make `serverScanRequest` public?

Reply:
Yes exactly! This would remove the additional burden. It's what most of our code does.